File: solutions/zig/darts/1/darts.zig
const std = @import("std");

pub const Coordinate = struct {
    x: f32,
    y: f32,

    pub fn init(x_coord: f32, y_coord: f32) Coordinate {
        return .{
            .x = x_coord,
            .y = y_coord,
        };
    }
    pub fn score(self: Coordinate) usize {
        const dist = std.math.sqrt(self.x * self.x + self.y * self.y);
        return if (dist <= 1.0) 10 else if (dist <= 5.0) 5 else if (dist <= 10.0) 1 else 0;
    }
};

File: solutions/go/darts/1/darts.go
package darts

func Score(x, y float64) int {
	dist := x*x + y*y
	if dist <= 1 {
		return 10
	} else if dist <= 25 {
		return 5
	} else if dist <= 100 {
		return 1
	} else {
		return 0
	}
}
